src/decorators/headerFooter.js

```diff
@@ -16,30 +17,25 @@ function headerFooter(ComposedComponent) {
         collapsed: props.collapsible && props.collapsible.collapsed
       };
 
-      this._headerElement = this._getElement({
-        type: 'header',
-        handleClick: props.collapsible ? this.handleHeaderClick : null
-      });
-
       this._cssClasses = {
         root: cx('ais-root', this.props.cssClasses.root),
         body: cx('ais-body', this.props.cssClasses.body)
       };
 
       this._footerElement = this._getElement({type: 'footer'});
     }
-    shouldComponentUpdate(nextProps, nextState) {
-      return nextState.collapsed === false ||
```

Comment:
We can keep that shouldComponentUpdate but we should do a deep comparison on the passed data to avoid re-rendering